Doc/c-api/gcsupport.rst

```diff
@@ -173,3 +173,43 @@ if the object is immutable.
    this method (don't just call :c:func:`Py_DECREF` on a reference).  The
    collector will call this method if it detects that this object is involved
    in a reference cycle.
+
+
+Controlling the Garbage Collector State
+---------------------------------------
+
+The C-API provides the following functions for controlling
+garbage collection runs.
+
+.. c:function:: Py_ssize_t PyGC_Collect(void)
+
+   Performs a garbage collection, if the garbage collector is enabled.
```

Comment:
```suggestion
   Performs a full garbage collection, if the garbage collector is enabled: same as calling func:`gc.collect`.
```

You may mention that free lists of built-in types are also cleared.

Reply:
It's actually not the same as `gc.collect()`. The latter doesn't care about `gc.enabled`, whereas `PyGC_Collect()` honours it.

Reply:
> I don't think this C-API function is the right place to document these details. That's what the gc module is for.

I agree that gc.collect() documentation must be the most complete. But in this case, mention that PyGC_Collect is similar to gc.collect() but just document changes between PyGC_Collect() and gc.collect(). As you wrote, the main one is that PyGC_Collect() honors gc.enabled.

Reply:
I can document `PyGC_Collect` later, let's focus this PR on the new APIs.

Reply:
I'm fine with documenting this "complex" function in a separated PR.

Reply:
I added a sentence to compare the two. Feel free to take another pass later.